dataformat-aml/src/test/java/io/adminshell/aas/v3/dataformat/aml/serialize/AmlSerializerTest.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2021 Fraunhofer-Gesellschaft zur Foerderung der angewandten Forschung e. V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.adminshell.aas.v3.dataformat.aml.serialize;
+
+import io.adminshell.aas.v3.dataformat.aml.fixtures.FullExample;
```

Comment:
Yes, you are right (in general). We are already working on this, however, FullExample is currently not 100% consistent acrosss different serialization formats. The reason for this is, that when implementing AML we  realized that the full example is actually not 100% valid according to the constraints but this has not caused issues in other serialization formats as they are more "error-friendly" e.g. unresolvable references do not cause JSON serializer to crash as references are just serialized as string whereas in AML unresolvable references can cause a crash because sometimes these references are resolved and the referenced element is inlined.
We are working on unifying FullExample.java but we postponed this task until validation is implemeted and we can be sure that we fix all issues in the FullExample in one go as this requires slight adjustments to the expected results of the unit tests of all serializers.